src/gennylib/include/gennylib/context.hpp

```diff
@@ -354,69 +354,23 @@ class ActorContext final : public v1::ConfigNode {
     // <Forwarding to delegates>
 
     /**
-     * Convenience method for creating a metrics::Timer.
+     * Convenience method for creating a metrics::Operation that's unique for this actor and thread.
      *
-     * @param operationName
-     *   the name of the thing being timed.
-     *   Will automatically add prefixes to make the full name unique
-     *   across Actors and threads.
+     * @param operationName the name of the operation being run.
      * @param id the id of this Actor, if any.
      */
-    auto timer(const std::string& operationName, ActorId id = 0u) const {
-        auto name = this->metricsName(operationName, id);
-        return this->_workload->_registry->timer(name);
-    }
-
-    /**
-     * Convenience method for creating a metrics::Gauge.
-     *
-     * @param operationName
-     *   the name of the thing being gauged.
-     *   Will automatically add prefixes to make the full name unique
-     *   across Actors and threads.
-     * @param id the id of this Actor, if any.
-     */
-    auto gauge(const std::string& operationName, ActorId id = 0u) const {
-        auto name = this->metricsName(operationName, id);
-        return this->_workload->_registry->gauge(name);
-    }
-
-    /**
-     * Convenience method for creating a metrics::Counter.
-     *
-     *
-     * @param operationName
-     *   the name of the thing being counted.
-     *   Will automatically add prefixes to make the full name unique
-     *   across Actors and threads.
-     * @param id the id of this Actor, if any.
-     */
-    auto counter(const std::string& operationName, ActorId id = 0u) const {
-        auto name = this->metricsName(operationName, id);
-        return this->_workload->_registry->counter(name);
-    }
-
     auto operation(const std::string& operationName, ActorId id = 0u) const {
-        auto name = this->metricsName(operationName, id);
-        return this->_workload->_registry->operation(name);
+        return this->_workload->_registry->operation(
+            id, this->get<std::string>("Name"), operationName);
```

Comment:
no strong opinion but maybe we should remove the `id=0u` default now. If Actors don't specify it in two threads they'll unintentionally create races. I'm trying to think of a use-case when you'd actually want the default and can't think of any.

Reply:
Done.